addons/stock/models/stock_location.py

```diff
@@ -95,6 +97,11 @@ def _compute_next_inventory_date(self):
             else:
                 location.next_inventory_date = False
 
+    @api.depends('location_id.warehouse_id', 'warehouse_view_ids')
+    def _compute_warehouse_id(self):
+        for loc in self:
+            loc.warehouse_id = loc.get_warehouse()
```

Comment:
Is it still useful to have a `get_warehouse` method ?
It seems we call this method only at two other places and could use the field `warehouse_id` instead now.